fairroot/geobase/FairGeoSet.h

```diff
@@ -89,10 +90,21 @@ class FairGeoSet : public TNamed
     TString& getAuthor() { return author; }
     TString& getDescription() { return description; }
     virtual const char* getKeepinName(Int_t, Int_t) { return 0; }
-    virtual const char* getModuleName(Int_t) { return 0; }
-    virtual const char* getEleName(Int_t) { return 0; }
+    virtual const char* getModuleName(Int_t);
+    virtual const char* getEleName(Int_t);
     virtual Int_t getSecNumInMod(const TString&) { return -1; }
-    virtual Int_t getModNumInMod(const TString&) { return 0; }
+    /** returns the module index in the modules array from the module name.
+        The module number is counted starting at 1, so to get the array index
+        one has to substract 1 from the module number.
+    */
+    virtual inline Int_t getModNumInMod(const TString& name)
+    {
+        // construct TString starting after the end of the detector name
+        // which contains the detector number
+        TString modNumber(name(fName.Length(), (name.Length() - fName.Length())));
```

Comment:
```suggestion
        assert(name.Length() > fName.Length());
        TString modNumber(name(fName.Length(), (name.Length() - fName.Length())));
```

Makes it a bit more readable IMHO, as this is really the assumption you are making, right?

Reply:
Done.